filebeat/inputsource/tcp/client.go

```diff
@@ -97,7 +99,7 @@ func (c *splitHandler) Handle(conn net.Conn) error {
 		if err != nil {
 			// we are forcing a Close on the socket, lets ignore any error that could happen.
 			select {
-			case <-c.done:
+			case <-closer.Done():
```

Comment:
if the splitter returns an error due to us closing the connection, shall we supress the error then and just return nil?

Reply:
This is currently what I do I suppress errors in that scenario, in the beats input, I've decided otherwise and it created a lot of confusion for the users.